.github/workflows/unit-test-on-pull-request.yml

```diff
@@ -137,6 +137,28 @@ jobs:
           name: integration-test-binaries-${{ matrix.target_arch }}
           path: support/*.test
 
+  coredump-test-macos:
+    name: Coredump tests (macOS)
+    runs-on: macos-latest
+    steps:
+      - name: Clone code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - name: Set up Go
+        uses: actions/setup-go@d35c59abb061a4a6fb18e82ac0862c26744d6ab5 # v5.5.0
+        with:
+          go-version-file: go.mod
+          cache-dependency-path: go.sum
+      - name: Cache coredump modules
+        uses: actions/cache@5a3ec84eff668545956fd18022155c47e93e2684 # v4.2.3
+        with:
+          path: tools/coredump/modulecache
+          key: coredumps-macos-${{ hashFiles('tools/coredump/testdata/*/*.json') }}
```

Comment:
Does the `key` need to be separate from others? I think this could be shared, because only the actual downloaded files are in the cache. There's nothing linux/mac specific.

Reply:
Thanks. Replaced the key with `arm64` to be shared.

Reply:
I don't fully understand the `restore-keys` options though. It feels like they are never used